clicache/src/Cache.cpp

```diff
@@ -55,6 +54,20 @@ namespace Apache
         m_nativeptr = gcnew native_shared_ptr<native::Cache>(nativeptr);
         m_pdxTypeRegistry = gcnew Apache::Geode::Client::Internal::PdxTypeRegistry(this);
         m_typeRegistry = gcnew Apache::Geode::Client::TypeRegistry(this);
+
+        m_disposedLock = gcnew Object();
+      }
+
+      Cache::~Cache() {
+        msclr::lock lockInstance(m_disposedLock);
+        if (!m_disposed) {
+          this->!Cache();
```

Comment:
We should be forcing invocation of `Close`, if it hasn't been called already, and eating any exceptions (maybe log but no exceptions should be thrown here) before deleting the `m_nativeptr`.